services/cd-service/pkg/service/batch.go

```
@@ -115,6 +115,51 @@ func ValidateApplication(
 	return nil
 }
 
+func ValidateEnvironment(
+	environmentName string,
+	environmentConfig config.EnvironmentConfig,
+) error {
+	if environmentConfig.ArgoCdConfigs != nil && environmentConfig.ArgoCd != nil {
```

this function should also return an error of both are nil